frame/pov-limit/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Pov Limit Pallet
+//!
+//! Pallet that consumes up to a specified weight of a block on `on_initialize`.
+//! The weight consumed is set as a percentage as a config parameter.
+//!
+//! NOTE: This is only meant to be used for testing
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Specifies the computation usage of this pallet on `on_initialize`.
+		#[pallet::constant]
+		type Compute: Get<u32>;
```

Comment:
Should be settable at runtime. Please have a look at the discussion in the forum https://forum.polkadot.network/t/push-kusama-limits-with-pov-weight-limit-system-parachains/1068